bedrock/src/test_utils.rs

```diff
@@ -0,0 +1,508 @@
+//! Test utilities for unit tests and E2E tests for mocking RPC responses either from Anvil or hard-coded for unit tests.
```

Comment:
Should this be in the main crate? And if it's only used for unit tests let's add a cfg directive or feature flag it if it needs to be used outside the crate